packages/samples/headless-react/src/components/relevance-inspector/relevance-inspector.fn.tsx

```diff
@@ -0,0 +1,47 @@
+import {useEffect, useState, FunctionComponent} from 'react';
+import {
+  buildRelevanceInspector,
+  RelevanceInspector as HeadlessRelevanceInspector,
+} from '@coveo/headless';
+import {engine} from '../../engine';
+
+interface RelevanceInspectorProps {
+  controller: HeadlessRelevanceInspector;
+}
+
+export const RelevanceInspector: FunctionComponent<RelevanceInspectorProps> = (
+  props
+) => {
+  const {controller} = props;
+  const [state, setState] = useState(controller.state);
+
+  useEffect(
+    () =>
+      controller.subscribe(() => {
+        console.info('Debug information [fn]', controller.state);
+        setState(controller.state);
+      }),
+    []
+  );
+
+  return (
+    <div>
+      <label>
+        Enable debug mode:{' '}
+        <input
+          type="checkbox"
+          checked={state.isEnabled}
+          onChange={() =>
+            state.isEnabled ? controller.disable() : controller.enable()
```

Comment:
Is the condition order correct here? When `isEnabled` is `true`, we disable the relevance inspector? Shouldn't it be the opposite?

Reply:
Yes, the order is correct, I tested it.

Reply:
I guess it just reads weird then? If the checkbox is enabled, we `disable` the inspector? 🤔

Reply:
when `onChange` is called, `state.isEnabled` isn't changed until `controller.disable()` or `controller.enable()` is called, which triggers a re-render.